### PR TITLE
Fix equalize_adapthist (CLAHE) for clip value 1.0, fix docstrings

### PR DESCRIPTION
**Pull request by funkey**
## Description

Addresses https://github.com/scikit-image/scikit-image/issues/4827 and fixes docstrings.

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Comment by funkey** [comment]
[This test](https://github.com/scikit-image/scikit-image/blob/master/skimage/exposure/tests/test_exposure.py#L479) fails now, which was authored by @rfezzani and @sciunto. Can you confirm that this is really the expected behavior?

**Comment by sciunto** [comment]
Why this test wouldn't be correct in your opinion?

**Comment by funkey** [comment]
Hi François,

I mentioned that briefly in https://github.com/scikit-image/scikit-image/issues/4827, but let me provide some more details here of how I understand the `clip_limit` parameter: From the docstring and the implementation, I assume that `clip_limit` is relative to the size of the kernel (i.e., the effective, absolute clip limit is `clip_limit * np.prod(kernel_size)`). A clip limit of 0.5, for example, would limit the histogram of intensities to at most 50% of all pixels in the kernel. Excess values in the histogram are evenly distributed across the other bins, hence performing the contrast limiting (CL) part of CLAHE. A clip limit of 1.0, as I understand it, would effectively disable clipping (no bin in the histogram can have more than 100% of all pixels). However, that doesn't mean that the input image remains unchanged. Adaptive histogram equalization is still performed (i.e., AHE), replacing the intensity of each pixel with its rank divided by kernel size.

This understanding is also in line with the progression of histogram equalization over clip limits close to 1.0. Compare, for instance, clip values of 0.9, 0.95, 1.0, and 1.05. The last value performs AHE in the current implementation. The first two values perform CLAHE with decreasing effect of clipping, i.e., in the limit AHE. In this sequence, I would find it surprising if 1.0 would not perform AHE, but instead return the original image.

**Comment by sciunto** [comment]
I'm afraid, as far as I remember, that we implemented the test after reading the misbehavior. It seems to me that you are correct, any other opinion @scikit-image/core ?

File: skimage/exposure/_adapthist.py [hunk 1]
```diff
@@ -119,8 +117,7 @@ def _clahe(image, kernel_size, clip_limit, nbins):
 
     The number of "effective" graylevels in the output image is set by `nbins`;
     selecting a small value (eg. 128) speeds up processing and still produce
```
Comment (mkcor) [inline skimage/exposure/_adapthist.py:119]:
```suggestion
    selecting a small value (e.g. 128) speeds up processing and still produces
```

File: skimage/exposure/_adapthist.py [hunk 2]
```diff
@@ -119,8 +117,7 @@ def _clahe(image, kernel_size, clip_limit, nbins):
...
-    an output image of good quality. The output image will have the same
-    minimum and maximum value as the input image. A clip limit smaller than 1
+    an output image of good quality. A clip limit of 0 or larger equal 1
```
Comment (mkcor) [inline skimage/exposure/_adapthist.py:120]: Is "a clip limit either == 0 or >=1" what is intended?

Comment (funkey) [reply skimage/exposure/_adapthist.py:120]: For CLAHE, the clip value would be in the open interval (0, 1). Clip values >=1 essentially disable clipping and therefore simply perform AHE. A clip value of 0 does not really make sense, so the implementation falls back to AHE in this case.

**Comment by jni** [comment]
From a cursory glance, I agree with these changes and that the test should be updated. @funkey (:wave:) can you update the test accordingly?

**Comment by funkey** [comment]
I believe the tests are fixed, the CircleCI error seems unrelated to me. Let me know if I can do anything else!

**Comment by stefanv** [comment]
Thanks for investigating, @funkey!

File: skimage/exposure/_adapthist.py (hunk 2 above)
Comment (emmanuelle) [inline skimage/exposure/_adapthist.py:120]:
```suggestion
    an output image of good quality. A clip limit of 0 or larger than or equal to 1
```

**Comment by emmanuelle** [comment]
@funkey I opened a small PR in your fork with minor improvements suggested in the review of this PR.

**Review by emmanuelle — APPROVED** [review]
Thank you @funkey the PR now looks ready to me!

**Review by mkcor — APPROVED** [review]
I have left minor suggestions!

File: skimage/exposure/_adapthist.py (hunk 2 above)
Comment (mkcor) [reply skimage/exposure/_adapthist.py]: Thanks for the very clear response!

**Comment by jni** [comment]
@mkcor I'm a bit ambivalent about the suggestions so I decided to merge this so we can move on with the release. Minor wording changes in docstrings can come later. :blush: Thanks for reviewing!

**Comment by mkcor** [comment]
@jni for sure! :)

**Pull request merged**